Sources/StreamChat/StateLayer/Chat.swift

```diff
@@ -1039,8 +1042,9 @@ public class Chat {
     ///   - handler: The handler closure which is called when the event happens.
     ///
     /// - Returns: A cancellable instance, which you use when you end the subscription. Deallocation of the result will tear down the subscription stream.
-    public func subscribe<E>(toEvent event: E.Type, handler: @escaping (E) -> Void) -> AnyCancellable where E: Event {
-        eventNotificationCenter.subscribe(
+    @MainActor public func subscribe<E>(toEvent event: E.Type, handler: @escaping (E) -> Void) throws -> AnyCancellable where E: Event {
+        let cid = try self.cid
+        return eventNotificationCenter.subscribe(
```

Comment:
Since `cid` can be nil and the function needs to return `AnyCancellable` then I had to change the function to throwing and also to run on the `@MainActor` (async throws would not make sense here)

Reply:
Change it. Not beautiful but nicer that MainActor + throws.